calculation factorial

itiszella opened the pull request.

itiszella committed:
calculation factorial

itiszella committed:
Merge branch 'master' of https://github.com/js-tasks-ru/jsbasic-20201214_itiszella into master

jsru commented:
Поздравляю, все задачи решены абсолютно верно
```diff
Вычислить факториал
+ факториал 3 равен 6 
+ факториал 1 равен 1
+ факториал 5 равен 120 
+ факториал 0 равен 1
```

Добавляю преподавателя (@SinimaWath) для код-ревью, посмотрим что он скажет :)

itiszella committed:
validation username

jsru commented:
Поздравляю, все задачи решены абсолютно верно
```diff
Вычислить факториал
+ факториал 3 равен 6 
+ факториал 1 равен 1
+ факториал 5 равен 120 
+ факториал 0 равен 1
```
```diff
Допишите функцию валидации имени пользователя
+ если меньше 4 символов то false
+ если имя корректное, то true
+ если содержит пробелы, то false
```

Добавляю преподавателя (@SinimaWath) для код-ревью, посмотрим что он скажет :)

itiszella committed:
Merge branch 'master' of https://github.com/js-tasks-ru/jsbasic-20201214_itiszella into master

itiszella committed:
uppercase first symbol

itiszella committed:
Merge branch 'master' of https://github.com/js-tasks-ru/jsbasic-20201214_itiszella into master

jsru commented:
Поздравляю, все задачи решены абсолютно верно
```diff
Сделать первый символ заглавным
+ должна работать для пустой строки
+ должна делать первый символ заглавным
+ должна работать для строки из одной буквы
```
```diff
Допишите функцию валидации имени пользователя
+ если меньше 4 символов то false
+ если содержит пробелы, то false
+ если имя корректное, то true
```
```diff
Вычислить факториал
+ факториал 3 равен 6 
+ факториал 1 равен 1
+ факториал 0 равен 1
+ факториал 5 равен 120 
```

Добавляю преподавателя (@SinimaWath) для код-ревью, посмотрим что он скажет :)

itiszella committed:
check spam

itiszella committed:
Merge branch 'master' of https://github.com/js-tasks-ru/jsbasic-20201214_itiszella into master

jsru commented:
Поздравляю, все задачи решены абсолютно верно
```diff
Допишите функцию валидации имени пользователя
+ если имя корректное, то true
+ если меньше 4 символов то false
+ если содержит пробелы, то false
```
```diff
Проверка на спам
+ должна возвращать false для 'innocent rabbit'
+ должна возвращать true для '1XbeT now'
+ должна возвращать true для 'free xxxxx'
```
```diff
Вычислить факториал
+ факториал 3 равен 6 
+ факториал 5 равен 120 
+ факториал 1 равен 1
+ факториал 0 равен 1
```
```diff
Сделать первый символ заглавным
+ должна работать для пустой строки
+ должна делать первый символ заглавным
+ должна работать для строки из одной буквы
```

Добавляю преподавателя (@SinimaWath) для код-ревью, посмотрим что он скажет :)

itiszella committed:
truncate a string

jsru commented:
Поздравляю, все задачи решены абсолютно верно
```diff
Допишите функцию валидации имени пользователя
+ если меньше 4 символов то false
+ если содержит пробелы, то false
+ если имя корректное, то true
```
```diff
Усечение строки
+ должна не трогать короткую строку 
+ должна усекать строку 
```
```diff
Проверка на спам
+ должна возвращать false для 'innocent rabbit'
+ должна возвращать true для 'free xxxxx'
+ должна возвращать true для '1XbeT now'
```
```diff
Сделать первый символ заглавным
+ должна делать первый символ заглавным
+ должна работать для пустой строки
+ должна работать для строки из одной буквы
```
```diff
Вычислить факториал
+ факториал 0 равен 1
+ факториал 1 равен 1
+ факториал 3 равен 6 
+ факториал 5 равен 120 
```

Добавляю преподавателя (@SinimaWath) для код-ревью, посмотрим что он скажет :)

SinimaWath commented on 1-module/2-task/index.js:
```diff
@@ -10,8 +10,14 @@ function print(text) {
+   
+   
+   // ваш код...
+ }
```
1. Давай поправим отступы и уберем лишние пустые строки. Про стиль можешь узнать вот здесь: https://learn.javascript.ru/coding-style и https://learn.javascript.ru/ninja-code. 
Должно получиться примерно так:
```
if (...) {
  return false;
}

return true;
```
2. На лекции проходили, что можно не делать явное приведение типов в if блоке: `Boolean(name) == false`, стоит заменить на `!name`, такой подход принят в if блоках
3. `name.split(' ').length - 1 > 0`, можно заменить на функцию `includes`, строки.  Узнать можешь вот здесь: https://learn.javascript.ru/string

SinimaWath commented on 1-module/3-task/index.js:
```diff
@@ -4,5 +4,8 @@
 function ucFirst(str) {
+  let newStr = '';
+  if (!str) return str;
+  return newStr = str[0].toUpperCase() + str.slice(1);
```
Это считается плохим тоном, присваивать и возвращать значение оператора `=`. Тут тебе просто не нужно переменная `newStr`

SinimaWath commented on 1-module/3-task/index.js:
```diff
@@ -4,5 +4,8 @@
  */
 function ucFirst(str) {
+  let newStr = '';
+  if (!str) return str;
```
1. Не хватает отступов между логическами блоками кода. Выше скидывал ссылочку где можно почитать.

SinimaWath commented on 1-module/3-task/index.js:
```diff
@@ -4,5 +4,8 @@
+  let newStr = '';
+  if (!str) return str;
+  return newStr = str[0].toUpperCase() + str.slice(1);
   // ваш код...
```
Лишний коммент, убери его плиз. В коде не стоит оставлять бесполезных комментов.

SinimaWath commented on 1-module/4-task/index.js:
```diff
@@ -4,5 +4,7 @@
  * @returns {boolean}
  */
 function checkSpam(str) {
+  if (str.toLowerCase().includes('1xBet'.toLowerCase()) || str.toUpperCase().includes('XXX')) return true;
```
Зачемания только по стилю кода, сверху есть ссылочка

SinimaWath commented on 1-module/5-task/index.js:
```diff
@@ -5,5 +5,8 @@
  */
 function truncate(str, maxlength) {
+  let newStr = '';
+  if (str.length > maxlength) return newStr = str.slice(0, maxlength - 1) + "…";
```
1. Отступы нужны
2. не нужна `newStr

SinimaWath replied:
И давай переделаем на шаблонные строки, про которые я рассказывал на лекции

SinimaWath reviewed (changes requested):
Решения хорошие, но много необычных стилистических решений. Давай их поправим :)

itiszella commented:
Не понятно как использовать шаблонные строки в пятом задании

вс, 20 дек. 2020 г., 19:02 Vladik Tarasov <notifications@github.com>:

> *@SinimaWath* requested changes on this pull request.
>
> Решения хорошие, но много необычных стилистических решений. Давай их
> поправим :)
> ------------------------------
>
> In 1-module/2-task/index.js
> <https://github.com/js-tasks-ru/jsbasic-20201214_itiszella/pull/2#discussion_r546395086>
> :
>
> > +
> +  if(Boolean(name) == false || name.length < 4 || name.split(' ').length - 1 > 0)  return false;
> +     return true;
> +
> +
> +
> +   // ваш код...
> + }
>
>
>    1. Давай поправим отступы и уберем лишние пустые строки. Про стиль
>    можешь узнать вот здесь: https://learn.javascript.ru/coding-style и
>    https://learn.javascript.ru/ninja-code.
>    Должно получиться примерно так:
>
> if (...) {
>   return false;
> }
>
> return true;
>
>
>    1. На лекции проходили, что можно не делать явное приведение типов в
>    if блоке: Boolean(name) == false, стоит заменить на !name, такой
>    подход принят в if блоках
>    2. name.split(' ').length - 1 > 0, можно заменить на функцию includes,
>    строки. Узнать можешь вот здесь: https://learn.javascript.ru/string
>
> ------------------------------
>
> In 1-module/3-task/index.js
> <https://github.com/js-tasks-ru/jsbasic-20201214_itiszella/pull/2#discussion_r546395330>
> :
>
> > @@ -4,5 +4,8 @@
>   * @returns {string}
>   */
>  function ucFirst(str) {
> +  let newStr = '';
> +  if (!str) return str;
> +  return newStr = str[0].toUpperCase() + str.slice(1);
>
> Это считается плохим тоном, присваивать и возвращать значение оператора =.
> Тут тебе просто не нужно переменная newStr
> ------------------------------
>
> In 1-module/3-task/index.js
> <https://github.com/js-tasks-ru/jsbasic-20201214_itiszella/pull/2#discussion_r546395340>
> :
>
> > @@ -4,5 +4,8 @@
>   * @returns {string}
>   */
>  function ucFirst(str) {
> +  let newStr = '';
> +  if (!str) return str;
>
>
>    1. Не хватает отступов между логическами блоками кода. Выше скидывал
>    ссылочку где можно почитать.
>
> ------------------------------
>
> In 1-module/3-task/index.js
> <https://github.com/js-tasks-ru/jsbasic-20201214_itiszella/pull/2#discussion_r546395417>
> :
>
> > @@ -4,5 +4,8 @@
>   * @returns {string}
>   */
>  function ucFirst(str) {
> +  let newStr = '';
> +  if (!str) return str;
> +  return newStr = str[0].toUpperCase() + str.slice(1);
>    // ваш код...
>
> Лишний коммент, убери его плиз. В коде не стоит оставлять бесполезных
> комментов.
> ------------------------------
>
> In 1-module/4-task/index.js
> <https://github.com/js-tasks-ru/jsbasic-20201214_itiszella/pull/2#discussion_r546395457>
> :
>
> > @@ -4,5 +4,7 @@
>   * @returns {boolean}
>   */
>  function checkSpam(str) {
> +  if (str.toLowerCase().includes('1xBet'.toLowerCase()) || str.toUpperCase().includes('XXX')) return true;
>
> Зачемания только по стилю кода, сверху есть ссылочка
> ------------------------------
>
> In 1-module/5-task/index.js
> <https://github.com/js-tasks-ru/jsbasic-20201214_itiszella/pull/2#discussion_r546395529>
> :
>
> > @@ -5,5 +5,8 @@
>   * @returns {string}
>   */
>  function truncate(str, maxlength) {
> +  let newStr = '';
> +  if (str.length > maxlength) return newStr = str.slice(0, maxlength - 1) + "…";
>
>
>    1. Отступы нужны
>    2. не нужна `newStr
>
> ------------------------------
>
> In 1-module/5-task/index.js
> <https://github.com/js-tasks-ru/jsbasic-20201214_itiszella/pull/2#discussion_r546395560>
> :
>
> > @@ -5,5 +5,8 @@
>   * @returns {string}
>   */
>  function truncate(str, maxlength) {
> +  let newStr = '';
> +  if (str.length > maxlength) return newStr = str.slice(0, maxlength - 1) + "…";
>
> И давай переделаем на шаблонные строки, про которые я рассказывал на лекции
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/js-tasks-ru/jsbasic-20201214_itiszella/pull/2#pullrequestreview-556029716>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AO5TGGP5JIMFWURNUISBDH3SVYNY7ANCNFSM4VCFS6OQ>
> .
>

itiszella committed:
work on bugs module 1

jsru commented:
Поздравляю, все задачи решены абсолютно верно
```diff
Проверка на спам
+ должна возвращать false для 'innocent rabbit'
+ должна возвращать true для '1XbeT now'
+ должна возвращать true для 'free xxxxx'
```
```diff
Сделать первый символ заглавным
+ должна работать для пустой строки
+ должна делать первый символ заглавным
+ должна работать для строки из одной буквы
```
```diff
Допишите функцию валидации имени пользователя
+ если меньше 4 символов то false
+ если имя корректное, то true
+ если содержит пробелы, то false
```
```diff
Усечение строки
+ должна не трогать короткую строку 
+ должна усекать строку 
```
```diff
Вычислить факториал
+ факториал 1 равен 1
+ факториал 0 равен 1
+ факториал 3 равен 6 
+ факториал 5 равен 120 
```

Добавляю преподавателя (@SinimaWath) для код-ревью, посмотрим что он скажет :)

itiszella commented:
Вроде разобрался с 5таском

пн, 21 дек. 2020 г., 11:14 javascript-ru <notifications@github.com>:

> Поздравляю, все задачи решены абсолютно верно
>
> Проверка на спам+ должна возвращать false для 'innocent rabbit'+ должна возвращать true для '1XbeT now'+ должна возвращать true для 'free xxxxx'
>
> Сделать первый символ заглавным+ должна работать для пустой строки+ должна делать первый символ заглавным+ должна работать для строки из одной буквы
>
> Допишите функцию валидации имени пользователя+ если меньше 4 символов то false+ если имя корректное, то true+ если содержит пробелы, то false
>
> Усечение строки+ должна не трогать короткую строку + должна усекать строку
>
> Вычислить факториал+ факториал 1 равен 1+ факториал 0 равен 1+ факториал 3 равен 6 + факториал 5 равен 120
>
> Добавляю преподавателя (@SinimaWath <https://github.com/SinimaWath>) для
> код-ревью, посмотрим что он скажет :)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/js-tasks-ru/jsbasic-20201214_itiszella/pull/2#issuecomment-748835122>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AO5TGGO5QV5WEHWUBF6KHELSV37URANCNFSM4VCFS6OQ>
> .
>

itiszella committed:
work on bugs 5 task

jsru commented:
Поздравляю, все задачи решены абсолютно верно
```diff
Допишите функцию валидации имени пользователя
+ если содержит пробелы, то false
+ если меньше 4 символов то false
+ если имя корректное, то true
```
```diff
Сделать первый символ заглавным
+ должна работать для пустой строки
+ должна делать первый символ заглавным
+ должна работать для строки из одной буквы
```
```diff
Проверка на спам
+ должна возвращать false для 'innocent rabbit'
+ должна возвращать true для '1XbeT now'
+ должна возвращать true для 'free xxxxx'
```
```diff
Усечение строки
+ должна не трогать короткую строку 
+ должна усекать строку 
```
```diff
Вычислить факториал
+ факториал 5 равен 120 
+ факториал 0 равен 1
+ факториал 1 равен 1
+ факториал 3 равен 6 
```

Добавляю преподавателя (@SinimaWath) для код-ревью, посмотрим что он скажет :)
